src/main/java/uk/nickbdyer/tictactoe/players/PlayerFactory.java

```diff
@@ -20,23 +17,15 @@ public PlayerFactory(UserInterface userInterface) {
         this.ui = userInterface;
     }
 
-    public List<Player> create(GameType type) {
-        if (type == GameType.HvsH) {
-            return Arrays.asList(new Human(X, ui), new Human(O, ui));
-        } else if (type == GameType.HvsAi) {
-            return Arrays.asList(new Human(X, ui), slow(new DumbComputer(O)));
-        } else if (type == GameType.AivsH) {
-            return Arrays.asList(slow(new DumbComputer(X)), new Human(O, ui));
-        } else if (type == GameType.AivsAi) {
-            return Arrays.asList(slow(new DumbComputer(X)), slow(new DumbComputer(O)));
-        } else if (type == GameType.HvsPAi) {
-            return Arrays.asList(new Human(X, ui), slow(new PerfectComputer(O)));
-        } else if (type == GameType.PAivsH) {
-            return Arrays.asList(slow(new PerfectComputer(X)), new Human(O, ui));
-        } else if (type == GameType.PAivsPAi) {
-            return Arrays.asList(slow(new PerfectComputer(X)), slow(new PerfectComputer(O)));
+    public Player create(PlayerType type, Mark mark) {
```

Comment:
`PlayerType` is coupled (by meaning) to the amount of implementations of `Player` you have.
Here there is a clear 1 to 1 relationship from `PlayerType` to `Player` implementation.
Adding a new `Player` implementation will cause `PlayerType` to change.

I wonder if there is a way to bring those parts closer, making the addition and removal of `Player` implementations a simple change in a single place.